packages/odf/components/s3-browser/public-access-block/utils.tsx

```diff
@@ -0,0 +1,188 @@
+import * as React from 'react';
+import { GetPublicAccessBlockCommandOutput } from '@aws-sdk/client-s3';
+import { TFunction } from 'react-i18next';
+import {
+  TreeViewDataItem,
+  TextContent,
+  TextVariants,
+  Text,
+  Label,
+} from '@patternfly/react-core';
+import { BanIcon, ExclamationTriangleIcon } from '@patternfly/react-icons';
+
+export const SupportedConfig = {
+  BlockPublicPolicy: 'BlockPublicPolicy',
+  RestrictPublicBuckets: 'RestrictPublicBuckets',
+};
+
+export const areAllChildrenChecked = (checkedItems: Set<string>) =>
+  Object.values(SupportedConfig).every((child) => checkedItems.has(child));
+
+export const areSomeChildrenChecked = (checkedItems: Set<string>) =>
+  Object.values(SupportedConfig).some((child) => checkedItems.has(child));
+
+export const isNoChildChecked = (checkedItems: Set<string>) =>
+  Object.values(SupportedConfig).every((child) => !checkedItems.has(child));
+
+export const getConfigText = (t: TFunction) => ({
+  [SupportedConfig.BlockPublicPolicy]: t(
+    'Block public access to buckets and objects granted through new public bucket policies'
+  ),
+  [SupportedConfig.RestrictPublicBuckets]: t(
+    'Block public and cross-account access to buckets and objects through any public bucket policies'
+  ),
+});
+
+export const showModal = (
+  pabData: GetPublicAccessBlockCommandOutput,
+  checkedItems: Set<string>,
+  t: TFunction
+) => {
+  const configText = getConfigText(t);
+  const disabledConfig = [];
+
+  Object.entries(SupportedConfig).forEach(([key, value]) => {
+    if (
+      pabData?.PublicAccessBlockConfiguration?.[key] === true &&
+      !checkedItems.has(value)
+    )
+      disabledConfig.push(configText[value]);
+  });
+
+  return [disabledConfig.length > 0, disabledConfig];
+};
+
+const getLabel = (manage: boolean, checkedItems: Set<string>, t: TFunction) => {
+  let color;
+  let text;
+  let icon;
+  const variant = manage ? 'outline' : 'filled';
+  if (areAllChildrenChecked(checkedItems)) {
+    color = 'green';
+    text = t('Blocked all');
+    icon = <BanIcon />;
+  } else if (areSomeChildrenChecked(checkedItems)) {
+    color = 'gold';
+    text = t('Partially blocked');
+    icon = <ExclamationTriangleIcon />;
+  } else {
+    color = 'red';
+    text = t('Unblocked all');
+    icon = <ExclamationTriangleIcon />;
```

Comment:
Provided the color is red...
```suggestion
    icon = <RedExclamationCircleIcon />;
```

Reply:
icon is automatically adapting the right color (check the screenshots), no need for explicit `RedExclamationCircleIcon`, `YellowExclamationCircleIcon` etc.